packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,706 @@
+responsible for handling validation, with methods for setting and retrieving the validation configuration. This approach ensures that the state of
+valid and invalid values passed within the mapping element is effectively managed. The class obtains the name of the validity flag to communicate this
+information via a public API called `validity`.
+
+By deriving from the base, the mention options can validate the following conditions for the `mention-url` and `display-name` value in mapping element:
+
+1. `validateMappingTypes(mappings)`
+2. `validateUniqueURL(URL, URLType)`
+3. `validateNoMissingURL(mappings)`
+4. `validateNoMissingURL(mappings)`
+5. `validateURL(mappings)`
+
+_Note_: These are subject to change based on the property changes in the `nimble-mapping-mention-user` element.
+
+If any of the mention option is invalid as per the above validation, that particular option will render as an empty
+option in the list. This indicates the client that some of the option is wrongly configured and identify the
+validation details using the public API `validity`.
+
+The public API to determine the validity of the mention options are `checkValidity()` and `validity`.
+See the [API section](#user-mention-element-non-visible-configuration-element) for more details.
+
+#### 6. _nimble-rich-text-user-mention-view_:
+
+The foundation for configuring the rendering of mention nodes in the UI is provided by a base class known as `nimble-rich-text-mention-view`. This base class is an extension of
+`FoundationElement`.
+
+The `nimble-rich-text-user-mention-view` is a view class that is derived from the above base class, and its template contains a `span` element with the required CSS styling.
+They are equipped with attributes such as `mention-url`, `mention-label`, and `mention-type` to render the node as a `Mention` node in Tiptap editor.
+It uses the default slot to render the same text content from the editor to the view element's shadow root.
+
+The Mention node from Tiptap is extended, and the [`renderHTML`](https://tiptap.dev/guide/custom-extensions/#render-html) method is overridden to render
+the element as a `nimble-rich-text-user-mention-view` instead of the default `span` element as shown in the
+[Tiptap code](https://github.com/ueberdosis/tiptap/blob/42039c05f0894a2730a7b8f1b943ddb22d52a824/packages/extension-mention/src/mention.ts#L112).
+The attributes will be retained and included in the rendered mention nodes. The [`parseHTML`](https://tiptap.dev/guide/custom-extensions/#parse-html)
+will also hold the `nimble-rich-text-user-mention-view` tag and `span` with type attribute mention to load the specified tags as mention node
+in the editor.
+
+#### 7. _nimble-rich-text-mention-list-box_:
+
+This component has FAST's `list-box` in its template to inherit the essential mouse and keyboard interactions required within the dropdown list.
+To use the list box, a component named `nimble-list-box` will be created with just the styling changed from
```

Comment:
What would our plan be for how "official" to make the nimble-list-box? Will it have storybook documentation? Show up in the component table? If so, is it listed as incubating or done? Or is it just an internal element that clients aren't (yet) expected to use or even be aware of?

Reply:
We are planning to have it as an internal element that clients are not expected to use for now. This component is similar to the `nimble-list-option` we currently have without documentation in the storybook.